volume/drivers/nfs/nfs.go

```diff
@@ -211,14 +212,18 @@ func (d *driver) Create(
 	source *api.Source,
 	spec *api.VolumeSpec) (string, error) {
 
-	volumeID := locator.Name
-	if volumeID == "" && source.Parent != "" {
-		volumeID = d.getNewSnapVolID(source.Parent)
-		dlog.Infof("Creating snap vol id: %s", volumeID)
+	if locator.Name == "" {
+		return "", errors.New("volume name cannot be empty")
 	}
 
+	if hasSpaces := strings.Contains(locator.Name, " "); hasSpaces {
+		return "", errors.New("volume name cannot contain space characters")
+	}
+
+	volumeID := strings.TrimSuffix(uuid.New(), "\n")
```

Comment:
Why `\n` from the end? I do not think it has one.

Reply:
@lpabon I stole this line from buse driver. Maybe there was some reason the buse driver trimmed the newline?

Reply:
weird, yeah, maybe.